Source/Scene/MetadataUnpackingStep.js

```diff
@@ -0,0 +1,61 @@
+import Check from "../Core/Check.js";
+
+/**
+ * A namespace that contains several functions for producing GLSL code for
+ * unpacking values on the GPU. These functions take a GLSL expression as a
+ * string and wrap it with more GLSL code to create a new expression, much like
+ * a GLSL macro, but done in JS.
+ *
+ * @namespace MetadataUnpackingStep
+ *
+ * @private
+ */
+const MetadataUnpackingStep = {};
```

Comment:
The intent of these unpacking steps is to be composable. Right now I tried to make it so it generates one expression like so:

```
cast("int")(unnormalizeU8("x")) // => int(255.0 * (x))
```

The downside is if you have many steps, the generated line of code could look messy.

Another approach is to generate steps as new lines, something like this: (inspired by https://nodematerial-editor.babylonjs.com/)

```
// original
vec2 temp0 = texture2D(...).rg;
// after unnormalizeU8()
vec2 temp1 = 255.0 * temp0;
// after cast("int")
ivec2 temp2 = int(temp1);
// result
metadata.property = temp2;
```

This approach might allow for more complex initialization logic for things like arrays (see my other comment), but to do this correctly, I need to keep track of the types at every line. Doable, but it feels like unnecessary complexity. 

@lilleyse any thoughts here?